### ISSUE TITLE
An error occurred while attempting to add additional data to the stream.

### ISSUE THREAD

**Issue by arshidkv12** [karlseguin/smtp_client.zig#5]
An error occurred while attempting to add additional data to the stream.

**Error**: `expected client.AuthMode.CRAM_MD5, found null`

```
src/client.zig:419:9: 0x1059f219d in test.client: helo (test)
try t.expectEqual(.CRAM_MD5, client.ext_auth);
```

**Example Code:**

```zig
var stream = t.Stream.init();
        stream.add("220\r\n250-\r\n250 AUTH PLAIN LOGIN CRAM-MD5\r\n");
        var client = TestClient.init(stream, config);
        defer client.deinit();

        try client.hello();
        try t.expectEqual(false, client.ext_smtp_utf8);
        try t.expectEqual(false, client.ext_8_bit_mine);
        try t.expectEqual(.CRAM_MD5, client.ext_auth);
```

**Comment by karlseguin** [comment]
Not sure I understand the issue.

It looks like you're trying to run the tests, and getting a failure. But I can't reproduce it. What version of Zig?

Do you have any more data?

**Comment by arshidkv12** [comment]
I executed with the following command.

`zig build test --summary all`

**OS**: 20.5.0 Darwin Kernel Version 20.5.0

I think it is the issue with the `test_runner.zig` file. No error when I remove the `main` function from test_runner.zig.

**Comment by karlseguin** [comment]
If you remove `main` from test_runner.zig, then no tests are being run.

If you're running the tests, then this ought to be something you can debug. The trick would be to figure out if (a) `findSupportedAuth` is being called and, if so (b) why it's retuning null.

You can run just that test by doing:

TEST_FILTER="client: helo" zig build test

**Comment by arshidkv12** [comment]
I checked "findSupportedAuth" function. But data is not utf8.

**Comment by arshidkv12** [comment]
I tested in Linux server. Then I can see the same error.

**Comment by karlseguin** [comment]
Ok, I can reproduce this sometimes, looking into it

**Comment by karlseguin** [comment]
I believe this is fixed on master.

**Comment by arshidkv12** [comment]
Thank you

**Comment by karlseguin** [comment]
I just noticed that you're using 0.13. It's fixed in the zig-0.13 branch too.
